include/xsens_streaming/jointanglesdatagram.h

```diff
@@ -31,24 +31,60 @@
 
 class JointAnglesDatagram : public Datagram
 {
+public:
+  struct Joint
+  {
+    int32_t parent;
+    int32_t child;
+    float rotation[3];
+  };
 
 public:
   JointAnglesDatagram();
   virtual ~JointAnglesDatagram();
   virtual void printData() const override;
 
+  inline const std::vector<Joint> & data() const
+  {
+    return m_data;
+  }
+
 protected:
   virtual void deserializeData(Streamer & inputStreamer) override;
 
 private:
-  struct Joint
-  {
-    int32_t parent;
-    int32_t child;
-    float rotation[3];
-  };
-
   std::vector<Joint> m_data;
+
+  const std::vector<std::string> joint_names = {
+      "jL5S1",
+      "jL4L3",
+      "jL1T12",
+      "jT9T8",
+      "jT1C7",
+      "jC1Head",
+      "jT4RightShoulder",
+      "jRightShoulder",
+      "jRightElbow",
+      "jRightWrist",
+      "jT4LeftShoulder",
+      "jLeftShoulder",
+      "jLeftElbow",
+      "jLeftWrist",
+      "jRightHip",
+      "jRightKnee",
+      "jRightAnkle",
+      "jRightBallFoot",
+      "jLeftHip",
+      "jLeftKnee",
+      "jLeftAnkle",
+      "jLeftBallFoot",
+      "T8_Head",
+      "T8_LeftUpperArm",
+      "T8_RightUpperArm",
+      "Pelvis_T8",
+      "Vertical_Pelvis",
+      "Vertical_T8",
+  };
```

Comment:
Does this work when we use the MVN profiles with only the upper or lower body instead of the full body?

Reply:
That's a good remark, I'll check with xsens devs. We'll probably need to map joints from attached segments (e.g. jRightElbow is between Segment 9 (R Upper Arm) and Segment 10 (R Forearm)).